test/config/df-segment.data.xml

```diff
@@ -87,6 +87,16 @@
 
 <obj class="DFApplication" id="df-01">
  <attr name="application_name" type="string" val="daq_application"/>
+ <attr name="commandline_parameters" type="string">
+  <data val="--name"/>
+  <data val="df-01"/>
+  <data val="-c"/>
+  <data val="rest://localhost:3339"/><!-- ALTER_ME -->
+  <data val="-i"/>
+  <data val="kafka://monkafka.cern.ch:30092/opmon"/>
+  <data val="--configurationService"/>
+  <data val="oksconfig:test/config/test-session.data.xml"/><!-- ALTER_ME -->
```

Comment:
I know this is a hack to get things working but these these thing really do not belong here.  This not only adds many duplicated lines throughout the configuration but it also makes the Segment usable i only one Session. Why does drunc not construct the command line from OKS variables and its own command line?

Reply:
that would be the better option. For the future.

Reply:
The rationale for doing this was:
 - I don't want to modify the configuration on the fly in drunc.
 - I'm unsure if there are "placeholders" in OKS or if I should set some up myself.
I think the way to do this without adding too much complexity or changing the appfwk CLI is to add environment variables and set them up in the session (which I probably should have thought of earlier). This likely also touches on things discussed in the appfwk review.

I agree this isn't ideal.

For the command facility, it's a different problem: if it isn't set up correctly, then the parent controller doesn't know how to talk to their children (and therefore has to parse the CLI somewhere in drunc, which is not nice).